app/scripts/metamask-controller.js

```diff
@@ -1469,11 +1470,14 @@ export default class MetamaskController extends EventEmitter {
 
     // append origin to each request
     engine.push(createOriginMiddleware({ origin }))
+    // append tabId to each request if it exists
+    if (tabId) {
```

Comment:
Can we move this condition into the middleware itself?

Reply:
And then add a test for it

Reply:
I would think that having it within the middleware would be... worse? Better to skip it altogether if it's not needed

Reply:
Doesn't strike me as worse, but we can leave it as-is if you don't agree.

Still need a test for this case though, regardless.

Reply:
Sure, I can add a test

Reply:
I've added a test to ensure the `tabId` is omitted from the request if not provided (as opposed to being set to `null` or something). I didn't test for whether the middleware itself was constructed, as it didn't seem necessary to test how the internals of this module work.